jax/_src/api.py

```diff
@@ -2704,6 +2704,16 @@ def device_get(x: Any):
 
 
 class ShapeDtypeStruct:
+  """A container for the shape, dtype, and other static attributes of an array.
+
+  ``ShapeDtypeStruct`` is often used in conjunction with :func:`jax.eval_shape`.
+
+  Args:
+    shape: a sequence of integers representing an array shape
+    dtype: a dtype-like object
+    naned_shape: (optional) a dictionary representing a named shape
```

Comment:
named_shape

Reply:
Can we not document `named_shape`? It's experimental and only partially implemented, and I'm not sure if it'll stick around forever.

Reply:
In other words, I don't think it needs to be in the public API (though I may be mistaken).

Reply:
Done, thanks!